codalab/worker/docker_image_manager.py

```diff
@@ -151,14 +148,11 @@ def image_availability_state(image_spec, success_message, failure_message):
                 image = self._docker.images.get(image_spec)
                 digests = image.attrs.get('RepoDigests', [image_spec])
                 digest = digests[0] if len(digests) > 0 else None
-                with self._lock:
-                    self._image_cache[digest] = ImageCacheEntry(
-                        id=image.id,
-                        digest=digest,
-                        last_used=time.time(),
-                        virtual_size=image.attrs['VirtualSize'],
-                        marginal_size=image.attrs['Size'],
-                    )
+                timestamp = str(time.time())
+                for tag in image.tags:
+                    if tag.split(":")[0] == "codalab-image-cache/last-used":
+                        self._docker.images.remove(tag)
+                image.tag("codalab-image-cache/last-used", tag=timestamp)
```

Comment:
Should we put this line above the removal, so if something fails, we don't end up with a situation where an image that should be managed by codalab falls off the radar?

Reply:
Good point, I wanted to avoid deleting the new tag, but that's easily avoidable with a string comparison.